src/libcmd/installables.cc

```diff
@@ -1053,9 +1040,12 @@ StorePathSet Installable::toDerivations(
         for (const auto & b : i->toDerivedPaths())
             std::visit(overloaded {
                 [&](const DerivedPath::Opaque & bo) {
-                    if (!useDeriver)
-                        throw Error("argument '%s' did not evaluate to a derivation", i->what());
-                    drvPaths.insert(getDeriver(store, *i, bo.path));
+                    drvPaths.insert(
+                        bo.path.isDerivation()
+                            ? bo.path
```

Comment:
This allows `nix show-derivation /nix/store/asdfasdfasdf-foo.drv` to work. It would be silly to require `^*` for that!

If we do end up doing all of #7261, I am pretty sure the `useDriver` flag, and possibly this entire function, will disappear altogether.